packages/shared/watchOnce/index.ts

```diff
@@ -1,23 +1,38 @@
 import type { WatchCallback, WatchOptions, WatchSource, WatchStopHandle } from 'vue'
 import type { MapOldSources, MapSources } from '../utils'
-import { nextTick, watch } from 'vue'
+import { watch } from 'vue'
 
 // overloads
-export function watchOnce<T extends Readonly<WatchSource<unknown>[]>, Immediate extends Readonly<boolean> = false>(source: [...T], cb: WatchCallback<MapSources<T>, MapOldSources<T, Immediate>>, options?: WatchOptions<Immediate>): WatchStopHandle
+export function watchOnce<T extends Readonly<WatchSource<unknown>[]>>(
+  source: [...T],
+  cb: WatchCallback<MapSources<T>, MapOldSources<T, true>>,
+  options?: Omit<WatchOptions<true>, 'once'>
+): WatchStopHandle
 
-export function watchOnce<T, Immediate extends Readonly<boolean> = false>(sources: WatchSource<T>, cb: WatchCallback<T, Immediate extends true ? T | undefined : T>, options?: WatchOptions<Immediate>): WatchStopHandle
+export function watchOnce<T>(
+  source: WatchSource<T>,
+  cb: WatchCallback<T, T | undefined>,
+  options?: Omit<WatchOptions<true>, 'once'>
+): WatchStopHandle
 
-// implementation
-export function watchOnce<Immediate extends Readonly<boolean> = false>(
-  source: any,
-  cb: any,
-  options?: WatchOptions<Immediate>,
-): WatchStopHandle {
-  const stop = watch(source, (...args) => {
-    nextTick(() => stop())
+export function watchOnce<T extends object>(
+  source: T,
+  cb: WatchCallback<T, T | undefined>,
+  options?: Omit<WatchOptions<true>, 'once'>
+): WatchStopHandle
 
-    return cb(...args)
-  }, options)
-
-  return stop
+/**
+ * Shorthand for watching value with { once: true }
+ *
+ * @see https://vueuse.org/watchOnce
+ */
+export function watchOnce<T = any>(source: T, cb: any, options?: Omit<WatchOptions, 'once'>) {
+  return watch(
+    source as any,
+    cb,
+    {
+      ...options,
+      once: true,
+    },
+  )
 }
```

Comment:
```suggestion
export function watchOnce<T = any, Immediate extends Readonly<boolean> = false>(source: T | WatchSource<T>, cb: any, options?: Omit<WatchOptions<Immediate>, 'once'>) {
  return watch(
    source,
    cb,
    {
      ...options,
      once: true,
    },
  )
}
```

Reply:
yeah, and to be as close as possible to native watch

Reply:
not sure if we need to, but maybe we want to overload like vue does: 

https://github.com/vuejs/core/blob/1f98a9c493d01c21befa90107f0593bc92a58932/packages/runtime-core/src/apiWatch.ts#L87-L140

Reply:
@OrbisK Shall we do the same with the other watch shortcuts I guess.

I wonder as well why we don't do rest parameters and `typeof watch` (with a tail-like helper like the ones in `type-fest` to overwrite just the last parameter) so we don't have to keep this up to date ever...

Reply:
we can do, but does this work with the generics and overloads?